### PR TITLE
Fix for issue4024

### PR DESCRIPTION
**Pull request by mdehoon**
Fixes issue #4024 both for the MacOSX and the Cairo backends.

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -771,19 +774,23 @@ def draw(self, renderer):
                 clean_line, ismath = textobj.is_math_text(line)
 
                 if textobj.get_path_effects():
-                    from matplotlib.patheffects import PathEffectRenderer
-                    textrenderer = PathEffectRenderer(
-                                        textobj.get_path_effects(), renderer)
+                    path_effects = textobj.get_path_effects()
+                    if textobj.get_usetex():
+                        ismath = "TeX"
+                    path, transform = renderer._get_text_path_transform(x, y, clean_line, prop, angle, ismath)
```
Comment (tacaswell) [inline lib/matplotlib/text.py]: pep8 is choking on this line.

Comment (mdehoon) [inline lib/matplotlib/text.py]: True. I have modified the line to follow the PEP8 conventions.

-Michiel.

--------------------------------------------
On Wed, 6/17/15, Thomas A Caswell <notifications@github.com> wrote:

 Subject: Re: [matplotlib] Fix for issue4024 (#4532)
 To: "matplotlib/matplotlib" <matplotlib@noreply.github.com>
 Cc: "mdehoon" <mjldehoon@yahoo.com>
 Date: Wednesday, June 17, 2015, 1:07 PM
 
 In lib/matplotlib/text.py:
 > @@ -771,19 +774,23 @@ def
 draw(self, renderer):
 >                  clean_line, ismath =
 textobj.is_math_text(line)
 >  
 >                  if textobj.get_path_effects():
 > -                    from matplotlib.patheffects import
 PathEffectRenderer
 > -                    textrenderer = PathEffectRenderer(
 > -                                       
 textobj.get_path_effects(), renderer)
 > +                    path_effects =
 textobj.get_path_effects()
 > +                    if textobj.get_usetex():
 > +                        ismath = "TeX"
 > +                    path, transform =
 renderer._get_text_path_transform(x, y, clean_line, prop,
 angle, ismath)
 
 pep8 is choking on this line.
 
 —
 Reply to this email directly or view
 it on GitHub.

**Comment by mdehoon** [comment]
I have modified the offending line to follow the PEP8 conventions.

**Comment by tacaswell** [comment]
failure looks like a transient, kicked to restart

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -749,8 +750,10 @@ def draw(self, renderer):
             if textobj._bbox_patch:
                 textobj._draw_bbox(renderer, posx, posy)
 
+            color = textobj.get_color()
+            color = colorConverter.to_rgb(color)
```
Comment (tacaswell) [inline lib/matplotlib/text.py]: Should this be `to_rgba`?

**Comment by tacaswell** [comment]
I am still not sure I fully follow what is going on here, but isn't this treating the symptom and not the disease (which is `PathEffectRenderer` being broken)?

**Comment by tacaswell** [comment]
See discussion at #4202 

I am strongly :-1: on merging this as it working around a bug in some of the backends which is affecting the code paths of the rest of the backends in a way that makes the code more complex.

**Comment by efiring** [comment]
I'm closing this based on the discussion above and in #4202.

**Pull request closed without merging**